aiida_common_workflows/workflows/relax_and_bands.py

```diff
@@ -0,0 +1,324 @@
+    """Validate the sub process class."""
+    try:
+        process_class = WorkflowFactory(value.value)
+    except exceptions.EntryPointError:
+        return f'`{value.value}` is not a valid or registered workflow entry point.'
+
+    if not inspect.isclass(process_class) or not issubclass(process_class, CommonRelaxWorkChain):
+        return f'`{value.value}` is not a subclass of the `CommonRelaxWorkChain` common workflow.'
+
+
+def validate_sub_process_class_b(value, _):
+    """Validate the sub process class."""
+    try:
+        process_class = WorkflowFactory(value.value)
+    except exceptions.EntryPointError:
+        return f'`{value.value}` is not a valid or registered workflow entry point.'
+
+    if not inspect.isclass(process_class) or not issubclass(process_class, CommonBandsWorkChain):
+        return f'`{value.value}` is not a subclass of the `CommonBandsWorkChain` common workflow.'
+
+
+class RelaxAndBandsWorkChain(WorkChain):
+    """
+    Workflow to carry on a relaxation and subsequently calculate the bands.
+    """
+
+    @classmethod
+    def define(cls, spec):
+        # yapf: disable
+        super().define(spec)
+        spec.input_namespace(
+            'seekpath_parameters',
+            help='Inputs for the seekpath to be passed to `get_explicit_kpoints_path`.',
+        )
+        spec.input(
+            'seekpath_parameters.reference_distance',
+            valid_type=orm.Float,
+            default=lambda: orm.Float(0.025),
+            serializer=to_aiida_type,
+            help='Reference target distance between neighboring k-points along the path in units 1/Å.',
+        )
+        spec.input(
+            'seekpath_parameters.symprec',
+            valid_type=orm.Float,
+            default=lambda: orm.Float(0.00001),
+            serializer=to_aiida_type,
+            help='The symmetry precision used internally by SPGLIB.',
+        )
+        spec.input(
+            'seekpath_parameters.angle_tolerance',
+            valid_type=orm.Float,
+            default=lambda: orm.Float(-1.0),
+            serializer=to_aiida_type,
+            help='The angle tollerance used internally by SPGLIB.',
+        )
+        spec.input(
+            'seekpath_parameters.threshold',
+            valid_type=orm.Float,
+            default=lambda: orm.Float(0.0000001),
+            serializer=to_aiida_type,
+            help='The treshold for determining edge cases. Meaning is different depending on bravais lattice.',
+        )
+        spec.input(
+            'seekpath_parameters.with_time_reversal',
+            valid_type=orm.Bool,
+            default=lambda: orm.Bool(True),
+            serializer=to_aiida_type,
+            help='If False, and the group has no inversion symmetry, additional lines are returned.',
+        )
+
+        namspac = spec.inputs.create_port_namespace('relax_inputs')
+        namspac.absorb(CommonRelaxInputGenerator.spec().inputs)
+        namspac['protocol'].non_db = True
```

Comment:
Instead of doing this manually, I would write a recursive function and call it:
```python
def set_port_non_db(port):
    """Set ``non_db =  True`` for the given port.

    .. note:: If ``port``, is a port namespace, the function will be called recursively on all sub ports.

    :param port: a port or port namespace.
    """
    if isinstance(port, PortNamespace):
        for subport in port.values():
            set_port_non_db(subport)
    elif isinstance(port, InputPort):
        port.non_db = True
```
Then in the `define` method, you simply call the method:
```python
set_port_non_db(spec.inputs)
```